kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/internal/KafkaProxyInitializerTest.java

```diff
@@ -325,6 +354,25 @@ void shouldInitialiseTlsChannel() {
         kafkaProxyInitializer.initChannel(channel);
 
         // Then
-        verify(channelPipeline).addLast(isA(SniHandler.class));
+        verify(channelPipeline).addLast(eq("sniResolver"), isA(SniHandler.class));
+        assertErrorHandlerAdded();
+    }
+
+    @Test
+    void testLoggingErrorHandlerPreventsExceptionPropagatingToChannel() {
+        EmbeddedChannel embeddedChannel = new EmbeddedChannel();
+        embeddedChannel.pipeline().addLast(new ChannelInboundHandlerAdapter() {
+            @Override
+            public void channelRead(ChannelHandlerContext ctx, Object msg) {
+                throw new RuntimeException("failed to handle message: " + msg);
+            }
+        });
+        embeddedChannel.pipeline().addLast(new KafkaProxyInitializer.LoggingInboundErrorHandler());
+        embeddedChannel.writeInbound("arbitrary");
+        assertThatCode(embeddedChannel::checkException).doesNotThrowAnyException();
+    }
+
+    private void assertErrorHandlerAdded() {
+        verify(channelPipeline).addLast(eq("loggingInboundErrorHandler"), isA(KafkaProxyInitializer.LoggingInboundErrorHandler.class));
```

Comment:
should we bind the test so closely to the name of the handler? I kinda see why you have this, I just wonder if its too brittle.

Reply:
For this handler we don't care so much, I introduced the handler names because it became hard in the test to verify that the plain TLS handler had been added because it's just an anonymous ChannelInboundHandlerAdapter and the verify complained that we'd had `2` of these added. But I can remove the name comparisons elsewhere.

Reply:
more arguments for extracting top level classes :)